### PR TITLE
Bug 1395353 - Display Livemark Feed

### PR DESCRIPTION
**Pull request by joeyg**
This PR changes the way Live Bookmarks are displayed in the Firefox iOS app. Instead of providing a link to the site, this PR downloads the Atom or RSS feed and displays the feed items in a list providing easy access to the feed items.

Notes:
 - I used FeedKit (https://github.com/nmdias/FeedKit) as part of this PR. It is under MIT license and seemed to be actively maintained. Open to suggestions here.
 - Live Bookmarks ("Livemarks" in the code) have a icon associated with them in the Firefox desktop app. Does anyone have a mobile version of this icon? https://hg.mozilla.org/mozilla-central/raw-file/3529b653ede2/browser/themes/osx/page-livemarks.png
 - This PR does not support subscribing to Live Bookmarks in the Firefox iOS app - I plan to do that work separately.
 - Feed items are not opened in Reader View - should they be?

### CODE REVIEW

File: Client/Frontend/Home/LivemarkPanel.swift [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+        if #available(iOS 10.0, *) {
+            tableView.refreshControl = refreshControl
+        } else {
+            tableView.addSubview(refreshControl)
+        }
+        
+        self.isLoading = true
+        
+        
+//        tableView.addGestureRecognizer(longPressRecognizer)
```
Comment (joeyg) [inline Client/Frontend/Home/LivemarkPanel.swift]: Is there an appropriate Long press menu option?

File: Client/Frontend/Home/LivemarkPanel.swift [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+            guard let url = feedItem.url else { return }
+            guard let title = feedItem.title else { return }
+            let author = feedItem.author?.name
+            let published = feedItem.datePublished
+            addItem(url: url, title: title, author: author, published: published)
+        }
+    }
+    
+    private func handleFailure() {
+        // TODO: Display error
```
Comment (joeyg) [inline Client/Frontend/Home/LivemarkPanel.swift]: Is there a standard error view in the app?

**Comment by jhugman** [comment]
This looks excellent. I've had a cursory look at the code, but would like to get to it on Monday.

I've added this to our triage for next week, to start looking at localization and graphic design.

I've also tagged our UX person to start that conversation.

While this won't make it into next release, we have begun the process to getting into a future process. It may take some time and have to overcome some hurdles, so many many thanks!

**Comment by jhugman** [comment]
Just looking at this now. From the code, I'm trying to work out how to test it. I can't subscribe to a Livemark from the client, but I should be able to do that from the Desktop and then sync one over. 

Is there an easier way?

**Comment by st3fan** [comment]
@joeyg @jhugman This is great initiative, but for a feature like this we do have to run this past our product team to find out if this is something we are willing to include in the product. It is not a light decision since it will result in maintenance, extra dependencies, testing, ux work, continuous improvements, etc.

I don't want to discourage you, since it is rare that we get patches like this, but I must say that it is not a given that this can land in the product.

Livemarks are an odd feature, and we have to find out if this is something that desktop is also willing to support in the longer term. Which is very unclear at this point.

**Comment by joeyg** [comment]
@jhugman - Re:Testing - That's correct, you have to sync a Livemark from desktop to test this feature. I had planned to add subscribing to a Livemark from mobile in a separate PR but maybe that is too confusing.

**Comment by joeyg** [comment]
@st3fan - Thanks for taking a look at the PR and I understand the product concerns. This PR comes from a desire to improve the app's subscription functionality and there are certainly numerous ways to do that. Do you know if this is an area that the Mozilla Product Team is interested in discussing? Hopefully a single vision can be implemented across desktop and mobile apps.

**Comment by st3fan** [comment]
@jhugman can you please create a Trello card in the backlog for this feature and bring it up in the next funnel meeting? Also please know that I have no objections. I just want it to be a calculated choice instead of a drive-by feature.

**Comment by st3fan** [comment]
Relevant: http://fieldguide.gizmodo.com/why-rss-feeds-still-beat-facebook-and-twitter-for-track-1800722740

**Comment by joeyg** [comment]
Updated PR to include Icons from @tecgirl. Thanks!

**Comment by joeyg** [comment]
Screenshot of PR with latest icons: 

simulator screen shot - iphone se - 2017-09-19 at 13 31 43

**Comment by farhanpatel** [comment]
based on stefan's comments on Bugzilla. This is not something we can land and support.

**Pull request closed without merging**